index.js

```diff
@@ -1,21 +1,15 @@
 var loaderUtils = require('loader-utils');
 var MessageFormat = require('messageformat');
 
-module.exports = function(content) {
-  var query = loaderUtils.parseQuery(this.query);
-  var locale = query.locale || 'en';
-  var messages = (Array.isArray(this.inputValue) && typeof this.inputValue[0] === 'object') ? this.inputValue[0] : this.exec(content, this.resource);
```

Comment:
It does, at least for WebPack 2. If you want, you can still add a check for whether the source is proper JSON or not, and then strip the JS wrapper if necessary before `JSON.parse`ing it. That would let you avoid the `this.exec` while still supporting the JSON loaders even as they are now.